### ISSUE TITLE
Need (internal?) tokens_split()

### ISSUE THREAD

**Issue by kbenoit** [quanteda/quanteda#1500]
This would resolve the inefficiency noted [here](https://github.com/quanteda/quanteda/issues/1445#issuecomment-440066936) but with a general solution. I actually started writing this based on the (internal) `tokens_segment()` but then went for a quicker fix since I only needed an internal hyphen solution for 
https://github.com/quanteda/quanteda/blob/0dc5b345b4370ca8bc90c8cd0ecd6e4eb687e995/R/tokens.R#L221-L225

Proposal:
```r
tokens_split(x, pattern, valuetype = c("glob", "regex", "fixed"),
  case_insensitive = TRUE, keep_pattern = FALSE)
```
which would be the opposite direction, otherwise similar behaviour, to `tokens_compound()`.

**Comment by kbenoit** [comment]
BTW here I implemented a hyphen-specific version for dfm features [here](https://github.com/quanteda/quanteda/blob/master/R/textstat_lexdiv.R#L283), probably about as efficiently it's possible to do in straight R.

But it's simpler for a dfm easier than for tokens because a) can use column indexing from matrices and b) feature order does not matter for dfm.

**Comment by kbenoit** [comment]
To Do: Reimplement the inefficient solution to #1498 once `tokens_split()` is working.

**Comment by kbenoit** [comment]
Solved in #1520 via `tokens_chunk()`.
